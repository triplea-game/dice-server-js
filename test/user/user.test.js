const user = require('../../src/user/user');

describe('The user module', () => {
  const router = {
    get: jest.fn(),
  };

  const getHandlerForRoute = (route) => {
    user(router);
    expect(router.get).toHaveBeenCalledWith(route, expect.any(Function));
    const result = router.get.mock.calls.find(call => call[0] === route);
    if (result) {
      return result[1];
    }
    throw `No Route with name ${route} was registered`;
  };

  const res = {
    render: jest.fn(),
  };

  it('should return the passed router instance', () => {
    expect(user(router)).toBe(router);
  });

  it('should trigger the correct render on /', () => {
    const handler = getHandlerForRoute('/');

    handler(undefined, res);

    expect(res).toMatchSnapshot();
  });

  it('should trigger the correct render on /verify', () => {
    const handler = getHandlerForRoute('/verify');

    handler({
      query: {
        token: Buffer.from(JSON.stringify({
          dice: [0, 1, 1, 2, 3, 5, 8, 13],
          date: 626644800,
          signature: 'VGhlcmUgYXJlIG90aGVyIHNlY3JldHM='
        })).toString('base64'),
      },
    }, res);

    expect(res).toMatchSnapshot();
  });

  it('should trigger the correct render on /verify with invalid params', () => {
    const handler = getHandlerForRoute('/verify');

    handler({}, res);

    expect(res).toMatchSnapshot();
  });

  it('should trigger the correct render on /register', () => {
    const handler = getHandlerForRoute('/register');

    handler({
      query: {
        email: 'itsame@mario.com',
        token: 'My4xNDE1OTI2NTM1ODk3OTMyMzg0NjI2NDMzOA==',
      },
    }, res);

    expect(res).toMatchSnapshot();
  });

  it('should trigger the correct render on /unregister', () => {
    const handler = getHandlerForRoute('/unregister');

    handler({
      query: {
        email: 'itsame@luigi.com',
      },
    }, res);

    expect(res).toMatchSnapshot();
  });
});
