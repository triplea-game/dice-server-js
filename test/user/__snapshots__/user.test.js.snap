// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The user module should trigger the correct render on / 1`] = `
Object {
  "render": [MockFunction] {
    "calls": Array [
      Array [
        "register.html",
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;

exports[`The user module should trigger the correct render on /register 1`] = `
Object {
  "render": [MockFunction] {
    "calls": Array [
      Array [
        "confirm-register.html",
        Object {
          "email": "itsame@mario.com",
          "token": "My4xNDE1OTI2NTM1ODk3OTMyMzg0NjI2NDMzOA%3D%3D",
        },
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;

exports[`The user module should trigger the correct render on /unregister 1`] = `
Object {
  "render": [MockFunction] {
    "calls": Array [
      Array [
        "unregister.html",
        Object {
          "email": "itsame@luigi.com",
        },
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;

exports[`The user module should trigger the correct render on /verify 1`] = `
Object {
  "render": [MockFunction] {
    "calls": Array [
      Array [
        "verify.html",
        Object {
          "date": "Thu, 08 Jan 1970 06:04:04 GMT",
          "dice": Array [
            0,
            1,
            1,
            2,
            3,
            5,
            8,
            13,
          ],
          "token": "eyJkaWNlIjpbMCwxLDEsMiwzLDUsOCwxM10sImRhdGUiOjYyNjY0NDgwMCwic2lnbmF0dXJlIjoiVkdobGNtVWdZWEpsSUc5MGFHVnlJSE5sWTNKbGRITT0ifQ%3D%3D",
        },
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;

exports[`The user module should trigger the correct render on /verify with invalid params 1`] = `
Object {
  "render": [MockFunction] {
    "calls": Array [
      Array [
        "verify.html",
        Object {
          "invalid": true,
        },
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;
